packages/component/src/BasicWebChat.js

```diff
@@ -62,11 +64,104 @@ const BasicWebChat = ({ className }) => {
+
+      // Generally, we allow up/down arrow keys on all elements captured here, except those handled by the user agent.
+      // For example, if it is on <select>, we will ignore up/down arrow keys. Also true for textbox with autocomplete.
+
+      // For some elements, user agent don't handle arrow keys when ALT key is hold, so we can still handle ALT + UP/DOWN keys.
+      // For example, user agent ignore ALT + UP/DOWN on <input type="text"> with content.
+      // Counter-example, user agent continue to handle ALT + UP/DOWN on <input type="number">.
+      if (tagName === 'INPUT') {
+        const { list, type, value } = target;
+
+        if (
+          type === 'button' ||
+          type === 'checkbox' ||
+          type === 'file' ||
+          type === 'image' ||
+          type === 'radio' ||
+          type === 'reset' ||
+          type === 'submit'
+        ) {
+          // These are buttons, up/down arrow keys are not handled by the user agent.
+          allowArrowKeys = true;
+        } else if (
+          type === 'email' ||
+          type === 'password' ||
+          type === 'search' ||
+          type === 'tel' ||
+          type === 'text' ||
+          type === 'url'
+        ) {
+          if (autocomplete || list) {
+            // "autocomplete" and "list" are combobox. Up/down arrow keys could be handled by the user agent.
+            allowArrowKeys = false;
+          } else if (altKey || !value) {
+            // If it has content, user agent will handle up/down arrow and it works like HOME/END keys.
+            // "altKey" can be used, user agent ignore ALT + UP/DOWN.
+            allowArrowKeys = true;
+          }
+        }
+      } else if (tagName === 'SELECT') {
+        // User agent handle up/down arrow keys for dropdown list.
+        allowArrowKeys = false;
+      } else if (tagName === 'TEXTAREA') {
+        if (!autocomplete && (altKey || !target.value)) {
+          // User agent handle up/down arrow keys for multiline text box if it has content or is auto-complete.
+          allowArrowKeys = true;
+        }
+      } else if (target.getAttribute('contenteditable') === 'true') {
+        if (altKey || !target.innerHTML) {
+          // "contenteditable" element works like <textarea> minus "autocomplete".
+          allowArrowKeys = true;
+        }
+      } else {
+        allowArrowKeys = true;
+      }
+
+      if (allowArrowKeys) {
+        if (key === 'ArrowDown') {
+          event.preventDefault();
+
+          // If a custom HTML control want to handle up/down arrow, we will prevent them from listening to this event to prevent bugs due to handling arrow keys twice.
+          event.stopPropagation();
+
+          scrollDown();
+        } else if (key === 'ArrowUp') {
+          event.preventDefault();
+
+          // If a custom HTML control want to handle up/down arrow, we will prevent them from listening to this event to prevent bugs due to handling arrow keys twice.
```

Comment:
```suggestion
          // If a custom HTML control wants to handle up/down arrow, we will prevent them from listening to this event to prevent bugs due to handling arrow keys twice.
```